### DISCUSSION TITLE
Result reports no longer scrapeable after upgrading from 3.1.1 -> current

### DISCUSSION THREAD

**Discussion by john-lauer73** [pytest-dev/pytest-html#798] [Q&A]
Hi There. I have an automation that scrapes the results of pytest runs with python to send slack messages and since upgrading to the latest version of pytest-html, the automation has failed. The content of the results table seems to be dynamically rendered by js in the browser, but when working with the file, those results are unavailable. Any thoughts to restore functionality and is this a desired change after the version 4.0.0 update?

**Comment by BeyondEvil** [comment]
> The content of the results table seems to be dynamically rendered by js in the browser, but when working with the file, those results are unavailable.

Correct. However, all the data used when rendering is available in the HTML-file [here](https://github.com/pytest-dev/pytest-html/blob/master/src/pytest_html/resources/index.jinja2#L115)

> Any thoughts to restore functionality

As fara as I know, that's not possible since the report is now "client-side" rendered.
I'm open to suggestions however.

> and is this a desired change after the version 4.0.0 update?

Yes.

Depending on what your automation does and how it does it - you can use selenium to render the report before scraping it.

This is what the test suite does.

**Comment by RonnyPfannschmidt** [comment]
The report log extension as well as junitxml has more stable output,
